custom/aaa/tasks.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import, unicode_literals
+
+from datetime import datetime
+
+from celery.task import task
+from django.db import connections
+
+from custom.aaa.models import AggregationInformation, Child
+
+
+def update_table(domain, slug, method):
+    window_start = AggregationInformation.objects.filter(
+        step=slug, aggregation_window_end__isnull=False
+    ).order_by('-created_at').values_list('aggregation_window_end').first()
+    if window_start is None:
+        window_start = datetime(1900, 1, 1)
+    else:
+        window_start = window_start[0]
+
+    window_end = datetime.utcnow()
+    agg_info = AggregationInformation.objects.create(
+        domain=domain,
+        step=slug,
+        aggregation_window_start=window_start,
+        aggregation_window_end=window_end,
+    )
+
+    # implement lock
+
+    agg_query, agg_params = method(domain, window_start, window_end)
+    with connections['aaa-data'].cursor() as cursor:
+        cursor.execute(agg_query, agg_params)
+    agg_info.end_time = datetime.utcnow()
```

Comment:
This seems a little dangerous to me since it could result in missing a few rows that get inserted between the first date and now. Is it possible to have the query return the max `inserted_at` which could be used for the end time?